### DISCUSSION TITLE
QPageScroller

### DISCUSSION THREAD

**Discussion by wasabismile** [zauberzeug/nicegui#2816] [Ideas / Feature Requests]
### Question

When the page scrolls to a certain position, the "Back to Top" button appears. Clicking the button will bring the page back to the top

**Comment by falkoschindler** [comment]
Hi @wasabismile,

Like other Quasar element without a native NiceGUI element, you can use the generic `ui.element` to wrap a `q-page-scroller` tag:
```py
for i in range(100):
    ui.label(f'Label {i}')

with ui.element('q-page-scroller'):
    ui.button(icon='arrow_upward')
```

<img width="681" alt="Screenshot 2024-04-04 at 07 45 01" src="https://github.com/zauberzeug/nicegui/assets/5767091/fa4e811c-b918-441e-aa46-0b2d630f16cb">

But I'll convert this discussion into a feature request, because a `ui.page_scroller` element would be nice to have. Anyone up for a pull request? 🙂

**Comment by wasabismile** [comment]
Thanks，I've tested your code, and it works very well! @falkoschindler 
But I have another question，How to implement the function of scrolling back to the object top  when using ui.scroll_area?
Here is my code, but it doesn't work.

```python
with ui.scroll_area().classes('bg-blue-100'):
    ui.label('CONTENT')
    [ui.label(f'Line {i}') for i in range(200)]
    with ui.element('q-page-scroller'):
         ui.button(icon='arrow_upward').props('round')
``` 

Please forgive me，I'm not a professional programmer. Maybe，my question is not professional enough

**Reply by falkoschindler** [reply]
No worries, your questions are perfectly fine.

QPageScroller only works for pages, not scroll areas. But you can create your one scroll-to-top button like this:
```py
with ui.scroll_area().classes('bg-blue-100') as scroll_area:
    ui.label('CONTENT')
    [ui.label(f'Line {i}') for i in range(200)]
    ui.button(icon='arrow_upward', on_click=lambda: scroll_area.scroll_to(percent=0)).props('round')
```

**Comment by wasabismile** [comment]
@falkoschindler  Thank you for your reply. Based on your suggestion, I tried using "QFab" to achieve the similar functionality, which enables scrolling back to the top and bottom. 
However, I found that I couldn't customize the style of QFab. It seems like I don't know how to set the style of the objects generated by using ui.element. For example, the expand direction and icons.

the following is my code:
```python
from nicegui import ui
with ui.scroll_area() as scroll_area: 
    ui.label('CONTENT')
    [ui.label(f'Line {i}') for i in range(2000)]
    with ui.element("q-fab") as upbtn:
        ui.element('q-fab-action').classes("bg-yellow").on("click",lambda: scroll_area.scroll_to(percent=0))
        ui.element('q-fab-action').classes("bg-blue").on("click",lambda: scroll_area.scroll_to(percent=1))
    upbtn.classes("fixed bottom-5 left-5 bg-accent text-white")
    ui.button(icon='arrow_upward', on_click=lambda: scroll_area.scroll_to(percent=0)).props('round').classes("fixed top-5 right-5")
ui.run()
```

**Reply by rodja** [reply]
In an `ui.element` all props must be set via `.props` (because there are no specalized parameters you could pass). `q-fab` and `q-fab-action` do need to be colorized with the `color` prop. Just yesterday I created a pull request to documents this: https://github.com/zauberzeug/nicegui/pull/2832.

**Reply by falkoschindler** [reply]
@wasabismile What exactly is not working?

**Reply by wasabismile** [reply]
@rodja @wasabismile , Thanks a lot. At the beginning, I didn't know how to set the icon props of q-fab and q-fab-action. Now , I learned.
```python
from nicegui import ui
with ui.scroll_area() as scroll_area: 
    ui.label('CONTENT')
    [ui.label(f'Line {i}') for i in range(2000)]
    with ui.element("q-fab",) as upbtn:
        down=ui.element('q-fab-action').classes("bg-blue").on("click",lambda: scroll_area.scroll_to(percent=1))
        upn=ui.element('q-fab-action').classes("bg-yellow").on("click",lambda: scroll_area.scroll_to(percent=0))
        
    upbtn.classes("fixed bottom-5 right-5 bg-accent text-white")
    upbtn.props('icon="keyboard_arrow_left"')
    upbtn.props('direction="up"')
    upbtn.props('padding="xs"')
    upn.props('icon="keyboard_arrow_up"')
    down.props('icon="keyboard_arrow_down"')
    ui.button(icon='arrow_upward', on_click=lambda: scroll_area.scroll_to(percent=0)).props('round').classes("fixed top-5 right-5")

ui.run()
```
